Assignment4/dreynes-180248/task07.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1hLS7IorY1ZXFRPbq9_a7fiPmY-lWHjr-
+
+**Task 07: Querying RDF(s)**
+"""
+
+#!pip install rdflib 
+#github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#") 
+q1 = prepareQuery('''
+  SELECT DISTINCT ?person WHERE {
+      ?person rdfs:subClassOf <http://somewhere#Person>.
+  }
+  ''',
+   initNs = { "rdfs": RDFS}
+)
+
+# Visualize the results
+
+for r in g.query(q1):
+  print(r.person)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+q2 = prepareQuery('''
+  SELECT DISTINCT ?subj
+  WHERE { 
+    {
+      ?subj rdf:type <http://somewhere#Person>. 
+    }
+    UNION {
+      ?prop rdfs:subClassOf <http://somewhere#Person>.
+      ?subj rdf:type ?prop
+    }
+  }
+  ''',
+   initNs={"rdfs": RDFS, "rdf": RDF}
+  )
+# Visualize the results
+for r in g.query(q2):
+     print(r)
+for s, p, o in g.triples((None, None, "http://somewhere#Person")):
+  print(s)
```

Comment:
missing the instances of subclasses

Reply:
You need to specify the property, or this will list things like properties with range person